File: app/src/main/java/com/jeeldobariya/passcodes/ui/AboutUsActivity.java
package com.jeeldobariya.passcodes.ui;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.view.WindowCompat;
import com.jeeldobariya.passcodes.databinding.ActivityAboutUsBinding;
import com.jeeldobariya.passcodes.utils.Constant;
// import com.jeeldobariya.passcodes.utils.Permissions;

public class AboutUsActivity extends AppCompatActivity {
  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    ActivityAboutUsBinding binding = ActivityAboutUsBinding.inflate(getLayoutInflater());
    setContentView(binding.getRoot());

    // Add event onclick listener
    addOnClickListenerOnButton(binding);

    // Make window fullscreen
    WindowCompat.setDecorFitsSystemWindows(getWindow(), false);
  }

  private void openBrowser(String url) {
    Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
    startActivity(browserIntent);
  }

  // Added all the onclick event listiners
  private void addOnClickListenerOnButton(ActivityAboutUsBinding binding) {
    binding.viewSecurityGuidelinesBtn.setOnClickListener(
        v -> {
          openBrowser(Constant.SECURITY_GUIDE_URL);
        });

    binding.viewChangeLogBtn.setOnClickListener(
        v -> {
          openBrowser(Constant.CHANGELOG_URL);
        });

    binding.viewLicenseBtn.setOnClickListener(
        v -> {
          openBrowser(Constant.LICENSE_URL);
        });
  }
}

File: app/src/main/kotlin/com/jeeldobariya/passcodes/ui/AboutUsActivity.kt
package com.jeeldobariya.passcodes.ui

import com.jeeldobariya.passcodes.databinding.ActivityAboutUsBinding
import com.jeeldobariya.passcodes.utils.Constant
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.core.net.toUri

class AboutUsActivity : AppCompatActivity() {

  private lateinit var binding: ActivityAboutUsBinding

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    binding = ActivityAboutUsBinding.inflate(layoutInflater)
    setContentView(binding.root)

    binding.toolbar.setNavigationOnClickListener {
      onBackPressedDispatcher.onBackPressed()
    }

    // Set up button click listeners
    setupButtonListeners()
  }

  private fun openBrowser(url: String) {
    val browserIntent = Intent(Intent.ACTION_VIEW, url.toUri())
    startActivity(browserIntent)
  }

  private fun setupButtonListeners() {
    binding.cardSecurityGuidelines.setOnClickListener {
      openBrowser(Constant.SECURITY_GUIDE_URL)
    }

    binding.cardReleaseNotes.setOnClickListener {
      openBrowser(Constant.RELEASE_NOTE_URL)
    }

    binding.cardLicense.setOnClickListener {
      startActivity(Intent(this, LicenseActivity::class.java))
    }

    binding.cardReportBug.setOnClickListener {
      openBrowser(Constant.REPORT_BUG_URL)
    }

    binding.cardTelegramCommunity.setOnClickListener {
      openBrowser(Constant.TELEGRAM_COMMUNITY_URL)
    }
  }
}